docs/user/troubleshooting.md

```diff
@@ -50,3 +54,286 @@ To download the SOS report, run the following command on the device:
+- Out of memory errors in system logs
+- Applications killed by OOM killer during updates
+
+#### Diagnosis
+
+Check memory usage and OOM events:
+
+```console
+flightctl console device/${device_name} -- dmesg | grep -i "killed\|oom"
+flightctl console device/${device_name} -- free -h
+```
+
+Monitor memory during update process:
+
+```console
+flightctl console device/${device_name} -- watch "free -h && echo '---' && ps aux --sort=-%mem | head -10"
+```
+
+#### Resolution
+
+1. **Configure memory monitoring** with appropriate thresholds:
+   ```yaml
+   resources:
+   - monitorType: Memory
+     samplingInterval: 30s
+     alertRules:
+     - severity: Warning
+       duration: 5m
+       percentage: 75
+     - severity: Critical
+       duration: 2m
+       percentage: 85
+   ```
+
+2. **Schedule updates during low-memory periods**:
+   ```yaml
+   updatePolicy:
+     updateSchedule:
+       at: "0 3 * * 0"  # Sunday 3 AM when fewer processes are running
+   ```
+
+3. **Stop non-essential services** before major updates:
+   ```yaml
+   # Use device lifecycle hooks
+   # /etc/flightctl/hooks.d/beforeupdating/stop-services.yaml
+   - run: /usr/bin/systemctl stop my-memory-intensive-service
+     timeout: 30s
+   ```
+
+### Update Policy Conflicts
+
+#### Symptoms
+
+- Updates scheduled but never execute
+- Policy evaluation errors in agent logs
+- Download policy satisfied but update policy is not
+
+#### Diagnosis
+
+Check update policy configuration and current time evaluation:
+
+```console
+flightctl get device/${device_name} -o yaml | grep -A 20 "updatePolicy:"
+flightctl console device/${device_name} -- date
+flightctl console device/${device_name} -- journalctl -u flightctl-agent | grep -i policy
+```
+
+#### Resolution
+
+1. **Verify time zone configuration**:
+   ```console
+   flightctl console device/${device_name} -- timedatectl status
+   ```
+
+2. **Test cron expressions** using online tools like [crontab.guru](https://crontab.guru/)
+
+3. **Add grace duration** to account for timing variations:
+   ```yaml
+   updatePolicy:
+     updateSchedule:
+       at: "0 4 * * 0"
+       startGraceDuration: "2h"  # Allow 2-hour window
+   ```
+
+### Debugging Resource Monitor Configuration
+
+#### Verify Monitor Active Status
+
+Check if resource monitors are properly configured and running:
+
+```console
+flightctl console device/${device_name} -- journalctl -u flightctl-agent | grep -i "resource\|monitor\|alert"
+```
+
+#### Test Resource Threshold Triggers
+
+Temporarily create high resource usage to test alert triggers:
+
+**Disk usage test** (use with caution):
+
+```console
+flightctl console device/${device_name} -- dd if=/dev/zero of=/tmp/testfile bs=1M count=100
+# Monitor alerts, then clean up:
+flightctl console device/${device_name} -- rm /tmp/testfile
+```
+
+**Memory usage test**:
+
+```console
+flightctl console device/${device_name} -- stress --vm 1 --vm-bytes 512M --timeout 60s
```

Comment:
might want to put a warning here to be careful about the size of the test. this would also require stress to be installed dnf install stress --transient